src/app/data-model-provider/MetadataTypes.h

```diff
@@ -87,13 +132,176 @@ enum class AttributeQualityFlags : uint32_t
 struct AttributeEntry
 {
     AttributeId attributeId;
+
+
+    /**
+     * Checks if readPrivilege has a value.
+     */
+     bool readPrivilegeHasValue() const { return (mask & ATTRIBUTE_MASK_READ_PRVLG_ANY_VALUE); }
+
+    /**
+     * Checks if writePrivilege has a value.
+     */
+     bool writePrivilegeHasValue() const { return (mask & ATTRIBUTE_MASK_WRITE_PRVLG_ANY_VALUE); }
+
+
+     /**
+      * Getter for readPrivilege.
+      */
+     Access::Privilege GetReadPrivilege() const 
+     {
+ 
+         if ( (mask &  ATTRIBUTE_MASK_READ_PRVLG_KVIEW_VALUE) ==  ATTRIBUTE_MASK_READ_PRVLG_KVIEW_VALUE ) 
+           return (Access::Privilege::kView);
+         else
+         if ( (mask &  ATTRIBUTE_MASK_READ_PRVLG_KPROXYVIEW_VALUE) ==  ATTRIBUTE_MASK_READ_PRVLG_KPROXYVIEW_VALUE ) 
+           return (Access::Privilege::kProxyView);
+         else
+         if ( (mask &  ATTRIBUTE_MASK_READ_PRVLG_KOPERATE_VALUE) ==  ATTRIBUTE_MASK_READ_PRVLG_KOPERATE_VALUE ) 
+           return (Access::Privilege::kOperate);
+         else
+         if ( (mask &  ATTRIBUTE_MASK_READ_PRVLG_KMANAGE_VALUE) ==  ATTRIBUTE_MASK_READ_PRVLG_KMANAGE_VALUE ) 
+           return (Access::Privilege::kManage);
+         else
+         if ( (mask &  ATTRIBUTE_MASK_READ_PRVLG_KADMINISTER_VALUE) ==  ATTRIBUTE_MASK_READ_PRVLG_KADMINISTER_VALUE ) 
+           return (Access::Privilege::kAdminister);
+ 
+         return(Access::Privilege::kView); // generally defaults to View if readable
```

Comment:
In theory we can have write-only attributes. None in the spec today, however unittestingcluster has one such thing. This is why we return optional here.